src/platform/site-wide/side-nav/components/SideNav.js

```diff
@@ -14,12 +15,30 @@ class SideNav extends Component {
 
   constructor(props) {
     super(props);
+    this.debouncedResize = debounce(250, this.updateIsDesktop);
     this.state = {
       active: false,
+      isDesktop: this.getDesktop(), // adding this to trigger re-render on window resize
       navItemsLookup: props.navItemsLookup,
     };
   }
 
+  getDesktop = () => {
+    return window.innerWidth > 768;
+  };
+
+  updateIsDesktop = () => {
+    this.setState({ isDesktop: this.getDesktop() });
+  };
```

Comment:
Would `getIsDesktop` and `setIsDesktop` be more consistent for naming?

Reply:
This could be a really useful React hook!